### PR TITLE
[FINERACT-1678] Fix Inline COB execution

### PR DESCRIPTION
**Pull request by taskain7**
## Description

Fixing the Inline Loan COB execution

### CODE REVIEW

File: fineract-provider/src/test/java/org/apache/fineract/cob/common/InitialisationTaskletStepDefinitions.java [hunk 1]
```diff
@@ -48,11 +59,22 @@ public InitialisationTaskletStepDefinitions() {
...
+            jobExecution.getExecutionContext().put(LoanCOBConstant.BUSINESS_DATE_PARAMETER_NAME,
+                    cobBusinessDate.format(DateTimeFormatter.ISO_DATE));
+            StepExecution stepExecution = new StepExecution("step", jobExecution);
+            StepContext stepContext = new StepContext(stepExecution);
+            chunkContext = new ChunkContext(stepContext);
         });
 
         When("InitialisationTasklet.execute method executed", () -> {
-            resultItem = this.initialisationTasklet.execute(null, null);
+            resultItem = this.initialisationTasklet.execute(null, chunkContext);
```
Comment (adamsaghy) [inline fineract-provider/src/test/java/org/apache/fineract/cob/common/InitialisationTaskletStepDefinitions.java:79]: Could you please enhanced this test to check whether the proper dates were provided and set?

File: fineract-provider/src/main/java/org/apache/fineract/cob/common/InitialisationTasklet.java [hunk 2]
```diff
@@ -41,11 +47,18 @@ public class InitialisationTasklet implements Tasklet {
...
     @Override
     public RepeatStatus execute(@NotNull StepContribution contribution, @NotNull ChunkContext chunkContext) throws Exception {
+        HashMap<BusinessDateType, LocalDate> businessDates = ThreadLocalContextUtil.getBusinessDates();
         AppUser user = userRepository.fetchSystemUser();
         UsernamePasswordAuthenticationToken auth = new UsernamePasswordAuthenticationToken(user, user.getPassword(),
                 new NullAuthoritiesMapper().mapAuthorities(user.getAuthorities()));
         SecurityContextHolder.getContext().setAuthentication(auth);
         ThreadLocalContextUtil.setActionContext(ActionContext.COB);
+        String businessDateString = Objects.requireNonNull((String) chunkContext.getStepContext().getStepExecution().getJobExecution()
+                .getExecutionContext().get(LoanCOBConstant.BUSINESS_DATE_PARAMETER_NAME));
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/cob/common/InitialisationTasklet.java:57]: I dont think the LoanCOBConstant.BUSINESS_DATE_PARAMETER_NAME got promoted. Would you please check it?

File: fineract-provider/src/test/java/org/apache/fineract/cob/common/InitialisationTaskletStepDefinitions.java [hunk 3]
```diff
@@ -48,23 +59,42 @@ public InitialisationTaskletStepDefinitions() {
...
         });
 
         Then("InitialisationTasklet.execute result should match", () -> {
             assertEquals(RepeatStatus.FINISHED, resultItem);
             assertEquals(appUser, SecurityContextHolder.getContext().getAuthentication().getPrincipal());
+            assertEquals(LocalDate.now(ZoneId.systemDefault()),
```
Comment (adamsaghy) [inline fineract-provider/src/test/java/org/apache/fineract/cob/common/InitialisationTaskletStepDefinitions.java]: Would you mind to please use a global variable for this? i know it is really edge case, but if this test got run at just before midnight, it might fail :D

**Review by adamsaghy — APPROVED** [review]
LGTM

**Review by galovics — APPROVED** [review]
